### ISSUE TITLE
NameError: name 'embedding_matrix' is not defined

### ISSUE THREAD

**Issue by stajilov** [oswaldoludwig/Seq2seq-Chatbot-for-Keras#18]
In conversation.py 
`    Shared_Embedding = Embedding(output_dim=word_embedding_size, input_dim=dictionary_size, weights=[embedding_matrix], input_length=maxlen_input, name='Shared')
`
however it produces the error above, should we reimport embeddings?

**Comment by oswaldoludwig** [comment]
Hi Vladimir,

If this line has been processed, it's because you don't have the weights_file, see the code:

### if os.path.isfile(weights_file):
    Shared_Embedding = Embedding(output_dim=word_embedding_size, input_dim=dictionary_size, input_length=maxlen_input, name='Shared')
else:
    Shared_Embedding = Embedding(output_dim=word_embedding_size, input_dim=dictionary_size, weights=[embedding_matrix], input_length=maxlen_input, name='Shared')
word_embedding_context = Shared_Embedding(input_context)

I suggest you download the net weights, "my_model_weights20.h5", which can be found [here](https://www.dropbox.com/sh/o0rze9dulwmon8b/AAA6g6QoKM8hBEHGst6W4JGDa?dl=0)

**Comment by stajilov** [comment]
yes, thanks, I collected my own corpus and retrained it, the training failed, that's why I didn't have this file and the respective error
